src/polyfills/stubs/intersection-observer-stub.js

```diff
@@ -23,6 +23,11 @@
  * amp-intersection-observer-polyfill extension.
  */
 
+/** @typedef {typeof IntersectionObserver} */
```

Comment:
IMHO, having `{typeof IntersectionObserver}` inlined where it is needed is preferable.
It is not reused in enough places to warrant an extra layer of indirection

Reply:
You're absolutely right; when I initially defined it, there were a lot more forced typecasts using it and all of the upgraders were using it as well. As I cleaned up types and identified that the Upgrader functions were common, turns out this was only used in one place (and only saves 2 characters after various stages of renaming lol). Updated here and ResizeObserver.